bindings/tflite/java/com/google/iree/native/tensor_wrapper.cc

```diff
@@ -0,0 +1,57 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "bindings/tflite/java/com/google/iree/native/tensor_wrapper.h"
+
+#include "bindings/tflite/java/com/google/iree/native/tflite_macros.h"
+#include "iree/base/logging.h"
+
+namespace iree {
+namespace tflite {
+
+TensorWrapper::TensorWrapper(const TfLiteTensor* tensor) : tensor_(tensor) {}
+
+TensorWrapper::TensorWrapper(TfLiteTensor* mutable_tensor)
+    : mutable_tensor_(mutable_tensor) {}
+
+TensorWrapper::~TensorWrapper() { tensor_ = nullptr; }
+
+iree_status_t TensorWrapper::CopyFromBuffer(const void* input_data,
+                                            size_t input_data_size) {
+  IREE_CHECK(is_mutable_tensor()) << "Can only copy into non-const tensors.";
```

Comment:
no checks (here and elsewhere) - always use status values (this should be INVALID_ARGUMENT)
I've been meaning to remove IREE_CHECK entirely - it should never be used